platform/kube/inject/initializer_test.go

```diff
@@ -237,3 +237,61 @@ func TestInitialize(t *testing.T) {
 		}
 	}
 }
+
+func TestModifyRequired(t *testing.T) {
+	cases := []struct {
+		namespace string
+		exclude   []string
+		ignore    []string
+		include   []string
+		want      bool
+	}{
+		{
+			namespace: "istio-system",
+			ignore:    []string{"istio-system"},
+			include:   []string{"b"},
+			want:      true,
+		},
+		{
+			namespace: "default",
+			ignore:    []string{"istio-system"},
+			include:   []string{"b"},
+			want:      false,
+		},
+		{
+			namespace: "istio-system",
+			ignore:    []string{"istio-system"},
+			include:   []string{"istio-system"},
+			want:      true,
+		},
+		{
+			namespace: "default",
+			ignore:    []string{"istio-system"},
+			exclude:   []string{"default"},
+			want:      true,
+		},
```

Comment:
Is this really true when default namespace is on he excluded list?

Reply:
I think this one makes sense... but I had qs is 2 tests below where namespace is foo-bar and foo-bar is on the included list.  Do we really want true there?

Reply:
Yes, the resource should be modified if it's namespace is on the include list. Another way to look at it is the resource should _always_ be modified unless it's namespace is not on the include, exclude, or ignore lists.